request.js

```diff
@@ -0,0 +1,41 @@
+const _request = require('request');
+
+const { ProtocolError } = require('./errors');
+
+/**
+ * Light wrapper around request.js. Returns a Promise and handles a couple
+ * Google-specific protocol interpretations.
+ *
+ * @param  {Object} options Forwarded to request.js
+ * @return {Promise} resolves with the request.js response
+ */
+const request = async (options) =>
+  new Promise((resolve, reject) => {
+    _request(options, (err, res) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      // Instead of using the `json` property in the request options, which
```

Comment:
To clarify one part: Why is this block also conditional on `!options.json`? Is it because, when specified, `request` has already taken care of body json part for us..?

Reply:
Ah yah -- let me update this comment, it's stale and obviously confusing. 

If `options.json=true` then request will (a) serialize the request body to JSON and set the `content-type` header _and_ automatically parse the response body to JSON.

We were already using `json=true` in some calls, but in others, like the request for a token, we needed to form-encode the request body and then parse a JSON response. So this guard tries to be compatible with JSON already being parsed. 

I think I originally have `typeof body === 'string'`, but of course JSON can deserialize to a string...

```js
> JSON.parse('"str"')
'str'
```